src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,121 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
+    private final RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+
+    @BeforeEach
+    void setUp() {
+        storageDao.add(new User(1L, "pavlo_ivanets", "458984", 18));
+        storageDao.add(new User(2L, "katerynka", "fks2045", 24));
```

Comment:
In the `setUp` method, the direct addition of users to `storageDao` bypasses the registration logic. This could lead to tests not accurately reflecting the registration process. Consider using `registrationService.register()` instead.